File: day19/test_day19.py
from unittest import TestCase
from day19.day19 import find_path_order, find_path_steps


class Day19Test(TestCase):
    def setUp(self) -> None:
        with open("day19/input.txt", "r") as file:
            self.inputFile = file.read().splitlines()
        with open("day19/input-test.txt", "r") as file:
            self.inputTestFile = file.read().splitlines()

    def test_find_path_order(self) -> None:
        self.assertEqual("ABCDEF", find_path_order(self.inputTestFile))
        self.assertEqual("AYRPVMEGQ", find_path_order(self.inputFile))

    def test_find_path_steps(self) -> None:
        self.assertEqual(38, find_path_steps(self.inputTestFile))
        self.assertEqual(16408, find_path_steps(self.inputFile))

File: day19/test-day19.lua
local lu = require('luaunit')
local File = require('lib.file')
local Day19 = require('day19.day19')

TestDay19 = {}

function TestDay19:setUp()
    self.input_test = File:lines_from('day19/input-test.txt')
    self.input = File:lines_from('day19/input.txt')
end

function TestDay19:test_count_beacons()
    lu.assertEquals(Day19.count_beacons(self.input_test), 79)
    lu.assertEquals(Day19.count_beacons(self.input), 400)
end

function TestDay19:test_find_greatest_distance()
    lu.assertEquals(Day19.find_greatest_distance(self.input_test), 3621)
    lu.assertEquals(Day19.find_greatest_distance(self.input), 12168)
end
